source/change-streams/change-streams.rst

```diff
@@ -347,16 +361,52 @@ Driver API
 
   class ChangeStreamOptions {
     /**
-     * Allowed values: ‘updateLookup’.  When set to ‘updateLookup’, the change notification
-     * for partial updates will include both a delta describing the changes to the document,
-     * as well as a copy of the entire document that was changed from some time after the
-     * change occurred. The default is to not send a value.
-     * For forward compatibility, a driver MUST NOT raise an error when a user provides an
-     * unknown value. The driver relies on the server to validate this option.
+     * Allowed values: 'updateLookup', 'whenAvailable', 'required'.
+     *
+     * The default is to not send a value. By default, the change notification
+     * for partial updates will include a delta describing the changes to the
+     * document.
+     *
+     * When set to 'updateLookup', the change notification for partial updates
+     * will include both a delta describing the changes to the document as well
+     * as a copy of the entire document that was changed from some time after
+     * the change occurred.
+     *
+     * When set to 'whenAvailable', configures the change stream to return the
+     * post-image of the modified document for replace and update change events
+     * if the post-image for this event is available.
+     *
+     * When set to 'required', the same behavior as 'whenAvailable' except that
+     * an error is raised if the post-image is not available.
+     *
+     * For forward compatibility, a driver MUST NOT raise an error when a user
+     * provides an unknown value. The driver relies on the server to validate
+     * this option.
+     *
      * @note this is an option of the `$changeStream` pipeline stage.
      */
     fullDocument: Optional<String>;
 
+    /**
+     * Allowed values: 'whenAvailable', 'required', 'off'.
+     *
+     * The default is to not send a value, which is equivalent to 'off'.
```

Comment:
Not our problem but curious why the server even has an explicit `off` option.

Reply:
Likely the same reason `fullDocument` originally supported a "default" string. IIRC, that was only required in pre-release versions. By the time change streams appeared in a stable release, "default" was equivalent to leaving it unspecified and there was no longer any purpose for the value.